### PR TITLE
Fix some spelling mistakes and receiver names, added dependabot for go modules

### PR DESCRIPTION
**Pull request by christian-kreuzberger-dtx**
Signed-off-by: Christian Kreuzberger <christian.kreuzberger@dynatrace.com>

Fixes some spelling mistakes and renamed some receivers as suggested by [go report card](https://goreportcard.com/report/github.com/keptn/go-utils).

Updated to gopkg.in/yaml.v2 v2.4.0 (same as already done in keptn/keptn, to avoid conflicts).

In addition, I've added dependabot for go modules and GH actions.

### CODE REVIEW

File: pkg/api/utils/apiUtils.go [hunk 1]
```diff
@@ -53,28 +53,28 @@ func (p *APIHandler) getHTTPClient() *http.Client {
 }
 
 // SendEvent sends an event to Keptn
-func (e *APIHandler) SendEvent(event models.KeptnContextExtendedCE) (*models.EventContext, *models.Error) {
+func (p *APIHandler) SendEvent(event models.KeptnContextExtendedCE) (*models.EventContext, *models.Error) {
```
Comment (warber) [inline pkg/api/utils/apiUtils.go:56]: why "p" ?

Comment (christian-kreuzberger-dtx) [reply pkg/api/utils/apiUtils.go:56]: This was suggested by GoLand, as all other receivers in this file are called `p

Comment (warber) [reply pkg/api/utils/apiUtils.go:56]: I think there are several conventions, in this case:
`a` or `ah` I also see `A` sometimes.
Maybe in the future, we'll have a list of coding guidelines/conventions for keptn development ?

Comment (christian-kreuzberger-dtx) [reply pkg/api/utils/apiUtils.go]: I've changed it to `a`

**Review by warber — APPROVED** [review]

**Pull request merged**